pkg/envoy/api/http_connection_manager.go

```diff
@@ -63,6 +63,8 @@ func NewHTTPConnectionManager(routeConfigName string, enableAccessLog bool) *hcm
 				RouteConfigName: routeConfigName,
 			},
 		},
+		//Force the connection manager to use the real remote address of the client connection.
+		UseRemoteAddress: &wrapperspb.BoolValue{Value: true},
```

Comment:
Could you set this `use_remote_address` only when proxy-protocol is enabled? It may be a breaking change for the existing users.